### DISCUSSION TITLE
How the LightManager works?

### DISCUSSION THREAD

**Discussion by inveders** [google/filament#4626] [Q&A] [answered]
Hi !
I have a white glb model load with my assetLoader in my code. I move my glb model in 3D space after I have define my lights like this:
```kotlin
fun defineLight(engine: Engine, scene: Scene?){
        // We now need a light, let's create a directional light
        val light = EntityManager.get().create()
        Log.d("debago","in define light 1")
        // Create a color from a temperature (5,500K)
        val (r, g, b) = Colors.cct(5_500.0f)
        LightManager.Builder(LightManager.Type.DIRECTIONAL)
            .color(r, g, b)
            // Intensity of the sun in lux on a clear day
            .intensity(110_000.0f)
            // The direction is normalized on our behalf
            .direction(0.0f, -0.5f, -1.0f)
            .build(engine, light)

        // Add the entity to the scene to light it
        scene?.addEntity(light)
    }
```
 When I am in front of my model, it's good, my model is white, but when I turn 180°, I see my model like I am behind it, and I see a black part of it. I think it's relative to the light I define. How can I have my model always in white, even if I turn 180° ? Do I have to remove light entity and put it again? How to set direction in the configuration? I don't understand how the light builder works. Thanks you
<img width="565" alt="Capture d’écran 2021-09-16 à 15 19 43" src="https://user-images.githubusercontent.com/40520645/133619528-1555006c-934e-4d37-a5ec-6df50936bcd8.png">

**Comment by romainguy** [comment, marked as answer]
You are using a directional light which is… well… directional :) To do what you want you need to add an `IndirectLight` to your scene (image-based lighting), which will provide illumination from all directions. You can check the Android samples to see how it's done, but the tl;dr is:

1. You need an environment map as a .hdr or .exr file (download one from [polyhaven](https://polyhaven.com/hdris))
2. You need to convert this environment map to a format Filament can use, with the `cmgen` command-line tool. Run this command: `cmgen -f ktx -x path/to/output path/to/map.exr`. You will get a `.ktx` file as an output
3. Put that `.ktx` file in your Android project (probably somewhere in `assets/`) and [use `KTXLoader` to load it as an `IndirectLight`](https://github.com/google/filament/blob/main/android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt#L126)

You'll want to choose an environment map (IBL) that matches what you expect a common environment to be for your users. For instance a [neutral indoor scene](https://polyhaven.com/a/photo_studio_01).

Once you have that working, know that ARCore can compute an environment map on the fly that matches the exact environment of the user. Those maps are however low resolution and may or may not be what you want.

**Reply by inveders** [reply]
Thank you, it was very useful to me !
